### DISCUSSION TITLE
VSCode Typescript ESLint (feat. Prettier) really slow to format

### DISCUSSION THREAD

**Discussion by okbytes** [react-hook-form/react-hook-form#5742] [Q&A] [answered]
**Context**
I've been refactoring one of our monster component pages to use `react-hook-form` across the whole component, which includes lots of sub-components that render conditional form bits based on the schema from our GQL API. I'm doing this cause the component is a pain to work with and so far, `react-hook-form` is _greatly_ simplifying the code and reasoning of the code.

**Issue**
I have Prettier running through TS-ESLint for super easy formatting stuff onSave. However, for this monster component (and it's sub-components), the whole thing is really slow and the VSCode toast notification confirms as much. Saving will trigger the ESLint stuff and then Prettier will do its thing. It takes about 2-5 seconds to save. Strangely, on any _other_ components in the app, things work normally and snappily. I suspect dumping a complex schema/defaultValues object for the form into `react-hook-form` is causing all the type-checking stuff to go nuts, but I haven't the first clue where to look for speed ups. To reiterate, this is about VSCode having trouble. The local dev code in the browser seems just fine as well as I _want_ `react-hook-form` to have this complex object for the form be managed by `react-hook-form`.

I welcome any rabbit trails or fresh ideas.

**Comment by okbytes** [comment, marked as answer]
I think it has something to do with `watch` cause I removed it from the top-level form parent and my VSCode performance is _much_ better.

**Reply by bluebill1049** [reply]
I am not sure why this would impact vscode, unless you have a browser install in your vs code, which trigger re-render with inputs update.

**Comment by Spoutnik97** [comment]
I have quite the same problem. But I suspect the Path type to be the cause

**Reply by Spoutnik97** [reply]
I saw discussions about the usage LazyPath, but it does seem to exist yet
